src/main/java/mate/academy/springboot/datajpa/dto/mapper/ProductMapper.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.datajpa.dto.mapper;
+
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMapper {
+    private final CategoryService categoryService;
+
+    public ProductMapper(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
```

Comment:
You do not need service in mapper. And id is enough for creating entityes connection in DB
```suggestion
```